core/templates/dev/head/pages/topic_editor/TopicEditorRoutingService.js

```diff
@@ -0,0 +1,65 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service that handles routing for the topic editor page.
+ */
+
+oppia.factory('TopicEditorRoutingService', [
+  '$rootScope', '$location', '$window', '$timeout', '$interval',
+  function(
+      $rootScope, $location, $window, $timeout, $interval) {
+    var MAIN_TAB = 'main';
+    var SUBTOPICS_TAB = 'subtopics';
```

Comment:
You know -- I've always thought that subtopics would be in the same tab as in the main tab. I.e. the main topic editor page contains a list of stories, followed by a list of skills (which have an interface similar to the Project page interface in GitHub which allows you to move skills freely between subtopics). It seems to me that skills (and their classification into subtopics) are an integral part of the topic, rather than something that should be relegated to a separate tab.

So ... what is the rationale for having subtopics on a separate tab, and what do you think of the above? Might be worth consulting with Meghan and @tjiang11 on this, too.

Reply:
I thought the add uncategorized skill to topic part can be done in the main editor itself, so including this there, would make the page long. Also, editing the html of a subtopic can also be done in the  same subtopics tab by hiding the subtopics list when editing.